ux.symfony.com/src/Twig/Components/Toolkit/ComponentDoc.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Twig\Components\Toolkit;
+
+use App\Enum\ToolkitKit;
+use Symfony\Component\HttpFoundation\UriSigner;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\UX\Toolkit\Component\Component;
+use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+use Tempest\Highlight\Highlighter;
+
+use function Symfony\Component\String\s;
+
+#[AsTwigComponent]
+class ComponentDoc
+{
+    public ToolkitKit $kit;
+    public Component $component;
+
+    public function __construct(
+        private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly UriSigner $uriSigner,
+        private readonly Highlighter $highlighter,
+    )
+    {
+
+    }
+
+    public function getContent(): string
+    {
+        return $this->formatContent($this->component->doc->markdownContent);
+    }
+
+    private function formatContent(string $markdownContent): string
+    {
+        $markdownContent = s($markdownContent);
+
+        // Decrease headings level (e.g.: "# Alert" to "## Alert", "## Examples" to "### Examples", etc...)
+        $markdownContent = $markdownContent->replaceMatches('/^(#{1,5})\s+(.+)$/m', '#$1 $2');
+
+        // Wrap each "```twig" code block with a HTML tab system containing the code and the component preview
```

Comment:
Is it possible there could be twig code that we don't want converted to this tab system?

Maybe this should be `twig+example`?

Reply:
Yes! I improved the component's doc rendering these two last days and it's already supported with "```twig {"preview":true}" (JSON is more easier to parse), but I didn't push yet :)

Reply:
Also: I wonder if some of this code could be simpler as a common mark extension?

Reply:
Yep, this is exactly what I did :D